src/core/include/mp-units/bits/sudo_cast.h

```diff
@@ -79,10 +81,56 @@ struct conversion_value_traits {
   static constexpr Magnitude auto num = numerator(M);
   static constexpr Magnitude auto den = denominator(M);
   static constexpr Magnitude auto irr = M * (den / num);
+  static constexpr auto ratio = [] {
+    if constexpr (std::is_integral_v<T>) {
+      using U = long double;
+      return detail::fixed_point<T>{get_value<U>(num) / get_value<U>(den) * get_value<U>(irr)};
+    } else {
+      return get_value<T>(num) / get_value<T>(den) * get_value<T>(irr);
+    }
+  }();
+  static constexpr bool value_increases = ratio >= T{1};
+
+  template<typename V>
+  static constexpr auto scale(V value)
+  {
+    if constexpr (std::is_integral_v<T>) {
+      return ratio.scale(value);
+    } else {
+      return value * ratio;
+    }
+  }
+};
+
+template<Magnitude auto M, typename T>
+  requires(is_integral(M))
+struct conversion_value_traits<M, T> {
+  static constexpr Magnitude auto num = numerator(M);
   static constexpr T num_mult = get_value<T>(num);
-  static constexpr T den_mult = get_value<T>(den);
-  static constexpr T irr_mult = get_value<T>(irr);
-  static constexpr T ratio = num_mult / den_mult * irr_mult;
+  static constexpr bool value_increases = true;
+
+  static_assert(get_value<T>(denominator(M)) == 1);
+  static_assert(is_integral(M));
```

Comment:
This is already checked in a constraint.

Reply:
Leftovers, from when I was figuring out #614 - I was perplexed of seeing a constexpr failure inside the  `_mul` case when the conversion factor between pounds and kilograms clearly is not a pure integral multiplication. However, it turned out to be a conversion to the "gcd" unit, which is chosen such that it is a pure multiplication, causing the overflow. I'll clean this up.